packages/beacon-node/src/api/impl/beacon/state/index.ts

```diff
@@ -137,7 +137,7 @@ export function getBeaconStateApi({
             balances.push({index: id, balance: state.balances.get(id)});
           } else {
             const index = headState.epochCtx.pubkey2index.get(id);
-            if (index != null && index <= state.validators.length) {
+            if (index !== undefined && index <= state.validators.length) {
```

Comment:
`headState.epochCtx.pubkey2index.get(id)` return type is `number | undefined` hence the compiler ensures `null` can never be returned. 

The code before the change works because when `index` is `undefined`, the `index != null` will return true (which is not what is expected), but `undefined <=  state.validators.length` will always return false.

The better way is to have strict checking against `undefined` which the diff introduces